syntax = "proto2";

package thumbnailer;

message ThumbnailerOption {
  // Desired (soft) maximum size limit in bytes.
  // This is what thumbnailer aims for at first. If it can't achieve this size
  // with at least 'min_lossy_quality', it will try to achieve 'hard_max_size'
  // instead.
  optional uint32 soft_max_size = 1 [default = 153600];

  // Hard limit for maximum file size. 
  // If it is less than 'soft_max_size', it will be set to 'soft_max_size'.
  optional uint32 hard_max_size = 2 [default = 153600];

  // Maximum PSNR change used in slope optimization.
  optional float slope_dpsnr = 3 [default = 1.0];

  // Number of times animation will loop.
  optional uint32 loop_count = 4 [default = 0];

  // Minimum lossy quality to be used for encoding each frame.
  optional uint32 min_lossy_quality = 5 [default = 0];

  // Effort/speed trade-off (0=fast, 6=slower-better).
  optional uint32 webp_method = 6 [default = 4];

  // If true, thumbnailer may use mixed lossy/lossless compression.
  optional bool allow_mixed = 7 [default = false];

  // If true, thumbnailer will print various encoding statistics.
  optional bool verbose = 8 [default = false];
}
